integration/common/openlineage/common/provider/dbt/structured_logs.py

```diff
@@ -265,6 +272,20 @@ def parse_node_finished_event(self, event) -> RunEvent:
             for model_input in self._get_model_inputs(node_unique_id)
         ]
         outputs = [self.node_to_output_dataset(node=self._get_model_node(node_unique_id), has_facets=True)]
+        if resource_type == "test":
+            success = node_status == "pass"
+            assertion = self._get_assertion(node_unique_id, success)
+            assertion_facet = data_quality_assertions_dataset.DataQualityAssertionsDatasetFacet(
+                assertions=[assertion]
```

Comment:
A notable difference here. There is a single asseration per test since there is only one assertion that is tested in every dbt test. Have a look at the unit [tests](https://github.com/MassyB/OpenLineage/blob/b7837beeb0154cf084cb1913c8d46688ae2eea24/integration/common/tests/dbt/structured_logs/postgres/events/results/successful_test_NodeFinished_OL.yaml#L226)

all the assertions that pertain to the same node (model) have the same input dataset.